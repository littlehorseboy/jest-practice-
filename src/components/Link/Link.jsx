import React from 'react';
import PropTypes from 'prop-types';

class Link extends React.Component {
  render() {
    return (
      <a href={this.props.page}>{this.props.children}</a>
    );
  }
}

Link.propTypes = {
  children: PropTypes.string.isRequired,
  page: PropTypes.string.isRequired,
};

export default Link;
