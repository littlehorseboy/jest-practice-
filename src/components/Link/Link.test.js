import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link.jsx';

it('Link component renders correctly', () => {
  const tree = renderer.create(
    <Link page="https://www.instagram.com">Instagram</Link>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
