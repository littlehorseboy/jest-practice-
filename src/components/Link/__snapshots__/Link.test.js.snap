// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders correctly 1`] = `
<a
  href="https://www.instagram.com"
>
  Instagram
</a>
`;
